cpp/include/ucxx/request_flush.h

```diff
@@ -0,0 +1,126 @@
+   *
+   * @param[in] endpointOrWorker    the parent component, which may either be a
+   *                                `std::shared_ptr<Endpoint>` or
+   *                                `std::shared_ptr<Worker>`.
+   * @param[in] requestData         container of the specified message type, including all
+   *                                type-specific data.
+   * @param[in] operationName       a human-readable operation name to help identifying
+   *                                requests by their types when UCXX logging is enabled.
+   * @param[in] enablePythonFuture  whether a python future should be created and
+   *                                subsequently notified.
+   * @param[in] callbackFunction    user-defined callback function to call upon completion.
+   * @param[in] callbackData        user-defined data to pass to the `callbackFunction`.
+   */
+  RequestFlush(std::shared_ptr<Component> endpointOrWorker,
+               const std::variant<data::Flush> requestData,
+               const std::string operationName,
+               const bool enablePythonFuture                = false,
+               RequestCallbackUserFunction callbackFunction = nullptr,
+               RequestCallbackUserData callbackData         = nullptr);
+
+ public:
+  /**
+   * @brief Constructor for `std::shared_ptr<ucxx::RequestFlush>`.
+   *
+   * The constructor for a `std::shared_ptr<ucxx::RequestFlush>` object, creating a request
+   * to flush outstanding AMO (Atomic Memory Operation) and RMA (Remote Memory Access)
+   * operations on a UCP endpoint or worker, returning a pointer to a request object that
+   * can be later awaited and checked for errors. This is a non-blocking operation, and its
+   * status must be verified from the resulting request object to confirm the flush
+   * operation has completed successfully.
+   *
+   * @throws ucxx::Error  `endpointOrWorker` is not a valid
+   *                      `std::shared_ptr<ucxx::Endpoint>` or
+   *                      `std::shared_ptr<ucxx::Worker>`.
+   *
+   * @param[in] endpointOrWorker    the parent component, which may either be a
+   *                                `std::shared_ptr<Endpoint>` or
+   *                                `std::shared_ptr<Worker>`.
+   * @param[in] requestData         container of the specified message type, including all
+   *                                type-specific data.
+   * @param[in] enablePythonFuture  whether a python future should be created and
+   *                                subsequently notified.
+   * @param[in] callbackFunction    user-defined callback function to call upon completion.
+   * @param[in] callbackData        user-defined data to pass to the `callbackFunction`.
+   *
+   * @returns The `shared_ptr<ucxx::RequestFlush>` object
+   */
+  friend std::shared_ptr<RequestFlush> createRequestFlush(
+    std::shared_ptr<Component> endpointOrWorker,
+    const std::variant<data::Flush> requestData,
```

Comment:
How come you are using a variant with a single type? Is this to follow a pattern?